### DISCUSSION TITLE
Home Assistant group switch

### DISCUSSION THREAD

**Discussion by lkankowski** [lkankowski/arduino-multi-relay2#28] [General]
@pepsonEL Pytanko mam odnośnie HA i problemu z masowym wyłączaniem - z którym wariantem mamy do czynienia:
1. niektóre urządzenia się nie wyłączają i tym samym w HA stan się nie zmienia
2. czy może tylko w HA nie ma widać zmiany, ale urządzenie się wyłączyło

Jak rozumiem, próbowałeś dokładnie tak jak było opisane w #13 i nie pomogło?

**Comment by pepsonEL** [comment]
Wariant 1.

**Comment by lkankowski** [comment]
@pepsonEL 
Jakbyś miał chwilę, to sprawdź wersję z brancha `loop-limit`.
https://github.com/lkankowski/arduino-multi-relay2/tree/loop-limit

Eksperymentalnie zrobiłem, że pętla `loop` obsługuje maks. 10 przycisków w jednym przebiegu, co w Twoim przypadku daje mikrokontrolerowi jakieś 6 razy więcej czasu na obsługę komunikacji szeregowej.

**Comment by pepsonEL** [comment]
Postaram się jutro sprawdzić albo w sobotę.... Dam znać.

**Comment by pepsonEL** [comment]
Niestety nic to nie dało. Jak opcja serial_RX_Buffer_size=1024 jest wyłaczona to nie gasi świateł. Tylko z tą opcją działa jakoś w miarę.

**Comment by lkankowski** [comment]
Szkoda. Myślałem, że w sprytny sposób załatwię sprawę.

**Comment by pepsonEL** [comment]
Ale i tak dzięki za chęci.

**Comment by szybki946** [comment]
Nie mogę wgrać do urządzenia 
![IMG_20220311_152536](https://user-images.githubusercontent.com/50736527/157891183-2655f858-e340-4f52-9f1c-aa3f4ba79bd1.jpg)
![IMG_20220311_152512](https://user-images.githubusercontent.com/50736527/157891323-3a1b2b89-4e76-4ba0-976c-41eeeb018ea7.jpg)

**Reply by lkankowski** [reply]
Widać, że nie przeczytałeś instrukcji, do czego zachęcam :)
musisz skopiować `config.h.sample` jako `config.h` i ustawić swoją konfigurację.

PS.
Jakbyś miał więcej pytań, to załóż osobną dyskusję, bo to nie ma nic wspólnego z HA, ani "group switch".

**Comment by lot-p** [comment]
Mi udało się rozwiązać włączanie/wyłączanie grupowe zmieniając prędkość transmisji serial port z 115200 na 9600. Nie wiem czy niesie to inne skutki ale ja nie zauważyłem.
Testy robiłem na HA@RaspberryPi4 z arduino 2560 32 przekaźniki oczywiście połączano Serial<>USB CH340 

U mnie jest to linia 60 w pliku main.cpp
`
```

// MySensors - This will execute before MySensors starts up
void before()
{
  Serial.begin(9600);

```
w HA też trzeba zmienić prędkość transmisji
